salt/states/boto3_route53.py

```diff
@@ -798,3 +826,204 @@ def rr_absent(name, HostedZoneId=None, DomainName=None, PrivateZone=False,
+        else:
+            ret['comment'] = 'Creation of Route 53 health check {} failed.'.format(Name)
+            log.error(ret['comment'])
+            ret['result'] = False
+            return ret
+    elif update:
+        keys = [
+            'HealthCheckVersion', 'IPAddress', 'Port', 'Type', 'ResourcePath', 'FullyQualifiedDomainName', 'SearchString',
+            'RequestInterval', 'FailureThreshold', 'MeasureLatency', 'Inverted', 'Disabled', 'HealthThreshold',
+            'EnableSNI', 'InsufficientDataHealthStatus', 'Regions', 'AlarmIdentifier', 'ChildHealthChecks', 'CallerReference'
+        ]
+
+        check = checks[0].get('HealthCheckConfig')
+        args = {'region': region, 'key': key, 'keyid': keyid, 'profile': profile}
+        changes = {key: params.get(key) for key in keys if params.get(key) is not None}
+
+        if len(reset_elements) > 0:
+            changes.update({'ResetElements': reset_elements})
+
+        if __opts__['test']:
+            ret['comment'] = 'Route 53 health check {} would be updated.'.format(Name)
+            keys = [key for key in keys if changes.get(key) is not None]
+            ret['changes'] = {key: changes.get(key) for key in keys if _ensure_unicode(changes.get(key)) != _ensure_unicode(check.get(key))}
+            ret['result'] = None
+            return ret
+
+        check = __salt__['boto3_route53.update_health_check'](check['Id'], changes, **args)
+        if check:
+            ret['comment'] = 'Route 53 health check {} successfully updated.'.format(Name)
+            log.info(ret['comment'])
+            ret['changes']['new'] = check
+            return ret
+
+    return ret
+
+def health_check_absent(name, Name=None, region=None, key=None, keyid=None, profile=None):
+    '''
+    Ensure the Route53 Health Check described is absent
+
+    name
+        The name of the state definition.
+    Name
+        The name of the health check.
+
+    region
+        The region to connect to.
+
+    key
+        Secret key to be used.
+
+    keyid
+        Access key to be used.
+
+    profile
+        Dict, or pillar key pointing to a dict, containing AWS region/key/keyid.
+    '''
+    Name = Name if Name else name
+
+    ret = {'name': name, 'result': True, 'comment': '', 'changes': {}}
+    args = {'Name': Name, 'region': region, 'key': key, 'keyid': keyid, 'profile': profile}
+
+    checks = __salt__['boto3_route53.find_health_checks'](Name=Name, region=region, key=key, keyid=keyid, profile=profile)
+
+    if len(checks) > 1:
+        ret['comment'] = 'Given criteria matched more than one health check.'
+        log.error(ret['comment'])
+        ret['result'] = False
+        return ret
+    elif len(checks) == 0:
+        ret['comment'] = 'Route 53 health check {} already absent'.format(Name)
+        log.info(ret['comment'])
+        return ret
+
+    if __opts__['test']:
+        ret['comment'] = 'Route 53 health check {} would be deleted.'.format(Name)
+        ret['result'] = None
```

Comment:
same question here on `None` vs `True`

Actually, I think it should have `ret['changes']['old'] = check and whatnot, too, like in the below block

Reply:
think this is still valid about the `ret['changes']['old']`. `None` is correct for test mode, though.